docs/tutorial.rst

```diff
@@ -306,17 +318,281 @@ trees. In this example we incorrectly inferred that 0 coalesces with the
+    
+    # URL for the VCF
+    url = (
+        "https://github.com/tskit-dev/tsinfer/raw/master/docs/_static/"
+        "P_dom_chr24_phased.vcf.gz")
+
+    with tsinfer.SampleData(path="P_dom_chr24_phased.samples") as samples:
+        vcf = cyvcf2.VCF(url)
+        add_diploid_sites(vcf, samples)
+    
+    print("Sample file created for {} samples ".format(samples.num_samples) +
+      "({} individuals) ".format(samples.num_individuals) +
+      "with {} variable sites.".format(samples.num_sites))
+
+    # Do the inference
+    ts = tsinfer.infer(samples)
+    print("Inferred tree sequence: {} trees over {} Mb ({} edges)".format(
+        ts.num_trees, ts.sequence_length/1e6, ts.num_edges))
+    
+On a modern computer, this should only take a few seconds to run, producing the
+following output::
+
+    Sample file created for 20 samples (20 individuals) with 13192 variable sites.
+    Inferred tree sequence: 6666 trees (32824 edges)
+
++++++++++++++++++++++++
+Adding more information
++++++++++++++++++++++++
+
+As the output indicates, we have created 20 individuals, each with a single chromosome,
+rather than (as we probably intended) 10 individuals each with 2 chromosomes. That's
+because the script has not called :meth:`.SampleData.add_individuals`, so
+``tsinfer`` has assumed that each input chromosome belongs to a single (haploid)
+:ref:`individual <sec_inference_data_model_individual>`. Although this is perfectly
+OK and does not invalidate further analysis, it may be difficult to match information
+from the original VCF file with nodes in the tree sequence. Ideally, you should aim to
+incorporate all necessary information into the sample data file, from where it will be
+carried through into the final inferred tree sequence. The code below demonstrates not
+only how to associate each pair of samples with a diploid individual, but also how to
+allocate an identifier to these individuals by using the individual's metadata field.
+The code also gives a basic illustration of adding extra information about the
+:ref:`population<sec_inference_data_model_population>` from which each individual has
+been sampled.
 
-    1. Add documentation links for msprime above so we can explain tree
-       sequences there.
+.. code-block:: python
 
-++++++++++++
-Data example
-++++++++++++
+    import json
+
+    def add_populations(vcf, samples):
+        # In this VCF, the first letter of the sample name refers to the population
+        pop_lookup = {'8': 'Norway', 'F': 'France'}
+        sample_populations = [pop_lookup[sample_name[:1]] for sample_name in vcf.samples]
+        name_to_pop_id = {}
+        for name in set(sample_populations):
+            pop_id = samples.add_population(metadata={'country': name})
+            name_to_pop_id[name] = pop_id
+        return {i: name_to_pop_id[name] for i, name in enumerate(sample_populations)}
+    
+    def add_diploid_individuals(vcf, samples, populations={}):
```

Comment:
Never use a mutable object as a function parameter [default](https://florimond.dev/blog/articles/2018/08/python-mutable-defaults-are-the-source-of-all-evil/).

This code should **only** work for the example we're giving and be absolutely no more general, so I don't think it should have any default.

Reply:
OK. Good tip, thanks.